tests/unit/test_kubernetes_control_plane.py

```diff
@@ -322,15 +322,16 @@ def test_setup_auth_webhook_tokens(kcs, ctsar):
     assert not is_flag_set("authentication.setup")
 
 
-@mock.patch.object(kubernetes_control_plane, "render", autospec=True)
-@mock.patch.object(kubernetes_control_plane, "get_ingress_address", autospec=True)
-def test_ignore_vip(get_ingress_address, render):
-    get_ingress_address.return_value = "5.6.7.8"
+@mock.patch("pathlib.Path.exists", mock.Mock(return_value=False))
+@mock.patch("pathlib.Path.glob", mock.Mock(return_value=[]))
+@mock.patch("pathlib.Path.mkdir", mock.Mock(return_value=0))
+def test_ignore_vip():
+    kubernetes_control_plane.any_file_changed.return_value = False
+    mock_gia = kubernetes_control_plane.get_ingress_address
+    mock_gia.return_value = "5.6.7.8"
```

Comment:
while testing with `python3.10` i noticed that this test wouldn't pass because mock.patch stated `kubernetes_control_plane.get_ingress_address` was already mocked.  Turns out that pre-python-3.10 this resulted in a mock of a mock -- so `get_ingress_address.assert_called_with` was a mock function and didn't actually test anything.  

Once unblocked by this double-mocking, i found the `register_auth_webhook(...)` hadn't been tested and needed to be updated to make it more testable.  This is the result of that work